### ISSUE TITLE
Distributed Contributions Are Ignored

### ISSUE THREAD

**Issue by pi0** [ossf/criticality_score#21]
Hi. I came across this project by discovering [this tweet](https://twitter.com/Vjeux/status/1337284323762835456).

It is very nice initiative but I think a real important factor ignored in current implementation which is many larger projects like [vue.js](https://github.com/vuejs), sperate parts like documentation and modular parts of frameworks in sperate repositories.

This leads to unfair and logically wrong scores since criticality of a project is defined by all of it's core components not by main repository (which is used as a hypothesis that 1 repo maps to 1 project)

This can go far behind for projects like [nuxt.js](https://github.com/nuxt/nuxt.js)([community](https://github.com/nuxt-community) + [contrib](https://github.com/nuxt-contrib)) and [webpack](https://github.com/webpack)([contrib](https://github.com/webpack-contrib)) and [rollup](https://github.com/rollup) which other than documentation, core components are developed in multiple repositories and organizations and each should be counted for commits, releases, issues, etc.

I would suggest supporting a special meta file in repositories for being able to define linked repositories:

Example: `nuxt/nuxt.js/.github/linked_repos`

```
nuxt/create-nuxt-app
nuxt/components
nuxt/vue-meta
nuxt/nuxtjs.org
nuxt/docs
```

**Comment by inferno-chromium** [comment]
Yes totally agree, we were just making a v1, kind of know the issue. Thinking how we can do this as automatic as possible.

**Comment by pi0** [comment]
Thanks for quick response @inferno-chromium :) One idea if I may suggest regarding automatic detection, would using dependency graph or package descriptors (like package.json) to include repositories on same org (or npm) scope. 

Yet it _might_ be tricky sometimes because we do not list something like documentation as a dependency or webpack-cli i think is an optional dependency.

**Comment by inferno-chromium** [comment]
@pi0 - so we dont use docs. in the current list of factors, which factors dont work for your repo, i guess releases, issue tracker, anything else ?

**Comment by pi0** [comment]
>  so we don't use docs. in the current list of factors,

Indeed but we implicitly use docs as a factor for projects that maintain documentation is same repository of project or using mono-repos which makes final score unbiased when compare two projects

> which factors dont work for your repo, i guess releases, issue tracker, dependent sub-repos, anything else ?

I think it would be perfect if for parameters below, we use **sum** from all linked repositories:

- `contributor_count`
- `org_count`
- `recent_releases_count`
- `closed_issues_count`
- `updated_issues_count`

Parameters below shall be only picked from main repository as is now : (each explained why)

- `created_since` (components may be introduced after project releaase)
- `updated_since` / `commit_frequency` (some utilities are extracted part of code. update time depends on project release itself)
- `dependents_count` (a dependent of sub-dependency is essentially dependent of main project so adding this makes double stats)

**Comment by inferno-chromium** [comment]
Need to check on popular packages missing e.g.: [atleast i can see custom issue tracker as an issue]
https://github.com/qt/qtbase
https://github.com/LibreOffice/core

**Comment by inferno-chromium** [comment]
@pi0 - just brainstorming here, i am thinking if you are looking for more org level ranking. I dont think we should combine projects for some org, and leave for others. but how about a ranking per org ? like https://github.com/vuejs, https://github.com/kubernetes. it does become hard in case like github.com/google
